pkg/image/api/types.go

```diff
@@ -60,6 +77,30 @@ type ImageLayer struct {
 	Size int64
 }
 
+// SignatureValidationResult holds a result of signature validation.
+type SignatureValidationResult string
+
+// ImageSignature holds a signature of an image.
+type ImageSignature struct {
```

Comment:
Sadly, GPG signatures do not carry full public keys, only key IDs, so we don’t automatically know the “certificate subject” (text describing the public key).

Of course if the public key is trusted per policy, the policy will have a copy of the key. Or we could ask a key server, and perhaps get a copy.